### ISSUE TITLE
How to compute your FID metric

### ISSUE THREAD

**Issue by LuigiSigillo** [2165998/TarGAN#4]
Hello there,
I am referring to #3 
I tried to use the model genesis approach, but I have some doubt about the correct implementation.
After the global avarege pooling in the last encoder layer, the tensor in output has a shape of (#batch_size, 512).
What's the continue to have the correct FID results ?
Do you perform some mean and that's the FID?
Am I missing something?

**Comment by cs-xiao** [comment]
Hi, once you get the tensors from model genesis approach, you can use these feature tensors to compute the FID by the following steps.

For example, compute the FID between modal FA and modal RA.

1. use the testing images of modal FA to get all the feature tensors  from model genesis approach, and concat them to output a shape of (N, 512), named **fea_fake**;
2. use the testing images of modal RA to get all the feature tensors  from model genesis approach, and concat them to output a shape of (N, 512), named **fea_real**;
3. then use the following codes to get the result

```
from scipy import linalg

def frechet_distance(mu, cov, mu2, cov2):
      cc, _ = linalg.sqrtm(np.dot(cov, cov2), disp=False)
      dist = np.sum((mu -mu2)**2) + np.trace(cov + cov2 - 2*cc)
      return np.real(dist)
  
mu.append(np.mean(fea_real.cpu().detach().numpy(), axis=0))
mu.append(np.mean(fea_fake.cpu().detach().numpy(), axis=0))
cov.append(np.cov(fea_real.cpu().detach().numpy(), rowvar=False))
cov.append(np.cov(fea_fake.cpu().detach().numpy(), rowvar=False))
  
fid_value = frechet_distance(mu[0], cov[0], mu[1], cov[1])
print(fid_value)
```

PS:  Merry Christmas~
